machine-learning-notebooks/transfer-learning-custom-azureml/docker/TFObjectDetectionAPI.dockerfile

```diff
@@ -0,0 +1,86 @@
+#############################################################################################
+# Copyright: Microsoft 2021
+# License: MIT
+# Purpose:  This Dockerfile is for use with Azure ML TF Object Detection experiments.
+# It's the base of an Environment image for training object detection models.
+# The base is from a build of the TensorFlow Object Detection API.
+# Additionally, the following is built into the image:
+# - Miniconda (custom Dockerfile method with Azure ML expects conda)
+# - Azure ML SDK
+#############################################################################################
+FROM nvidia/cuda:10.0-cudnn7-runtime
+
+ENV DEBIAN_FRONTEND noninteractive
+
+RUN apt-get update \
+      && apt-get install --no-install-recommends --no-install-suggests -y gnupg2 ca-certificates \
+            git build-essential libopencv-dev python3-opencv \
+      && rm -rf /var/lib/apt/lists/*
+
+RUN apt-get update \
```

Comment:
Is there are reason you split your RUN commands up here? I know it improves readability, but it also typically dramatically increases the size of Docker images.

Here's an interesting read about getting the size of Docker images down: https://learnk8s.io/blog/smaller-docker-images. I don't at all expect you to do more than just either use a multi-stage docker build or just combine some of these run statements. Or maybe you have a reason for having lots of run statements.

Reply:
Brilliant, thanks!  My build was soooooooo much faster. :)